File: Selenium/Activities/Activity13.py
from selenium import webdriver
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService 
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.common.by import By

service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service=service) as driver:
    driver.get("https://www.training-support.net/selenium/tables")
    print("The title of page is: ", driver.title)
    wait = WebDriverWait(driver,10)
    #Find the number of rows and columns in the table and print them.
    rows = driver.find_elements(By.XPATH,"//table[contains(@class, 'striped')]/tbody/tr")
    cols = driver.find_elements(By.XPATH,"//table[contains(@class, 'striped')]/tbody/tr[1]/td")
    print("Number of rows present: ", len(rows))
    print("Number of columns present are: ", len(cols))
#Find and print all the cell values in the third row of the table.
    third_row = driver.find_elements(By.XPATH,"//table[contains(@class, 'striped')]/tbody/tr[3]/td")
    print("cell values in third row are: ")
    for row in third_row:
        print("Cell value: ", row.text)
#Find and print the cell value at the second row second column.
    second_row_column_value = driver.find_element(By.XPATH,"//table[contains(@class, 'striped')]/tbody/tr[2]/td[2]")
    print("Cell value at second row and second column: ", second_row_column_value.text)

File: Selenium/Activities/activity13.java
/*Find and print the cell value at the second row second column.*/
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import java.util.List;

public class activity13 {
    public static void main(String[] args){
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://www.training-support.net/selenium/tables");
        System.out.println("The title of the page is: " + driver.getTitle());
        List<WebElement> rows = driver.findElements(By.xpath("//table[contains(@class, 'striped')]/tbody/tr"));
        System.out.println("Number of rows: " +rows.size());
        List<WebElement> cols = driver.findElements(By.xpath("//table[contains(@class, 'striped')]/tbody/tr[1]/td"));
        System.out.println("Number of columns: " + cols.size());
        List<WebElement> thirdRowvalues = driver.findElements(By.xpath("//table[contains(@class, 'striped')]/tbody/tr[3]/td"));
        System.out.println("Third row in the table is: ");
        for(WebElement cell: thirdRowvalues){
            System.out.println(cell.getText());
        }

        driver.quit();
    }
}
